android-components/components/service/contile/src/main/java/mozilla/components/service/contile/ContileTopSitesProvider.kt

```diff
@@ -160,24 +186,14 @@ class ContileTopSitesProvider(
     internal fun getBaseCacheFile(): File = File(applicationContext.filesDir, CACHE_FILE_NAME)
 }
 
-internal fun JSONObject.getTopSites(): List<TopSite.Provided> =
-    getJSONArray("tiles")
-        .asSequence { i -> getJSONObject(i) }
-        .mapNotNull { it.toTopSite() }
-        .toList()
-
-private fun JSONObject.toTopSite(): TopSite.Provided? {
-    return try {
-        TopSite.Provided(
-            id = getLong("id"),
-            title = getString("name"),
-            url = getString("url"),
-            clickUrl = getString("click_url"),
-            imageUrl = getString("image_url"),
-            impressionUrl = getString("impression_url"),
-            createdAt = null,
-        )
-    } catch (e: JSONException) {
-        null
+internal fun Headers.computeValidFor(): Long =
+    getAll("cache-control").sumOf {
+        val valueList = it.split("=")
+            .map { item -> item.trim() }
+
+        if (valueList.size == 2 && valueList[0] in listOf("max-age", "stale-if-error")) {
+            valueList[1].toLong()
+        } else {
+            0L
+        }
```

Comment:
nit: IMO it feels a little odd to split how we're handling extension functions in this package. My preference would be to move the `JSONObject` extensions back to this file since they don't appear to be used elsewhere, and it makes sense to test the extensions in the `ContileTopSitesProviderTest` as well. They seem to still be a single logical unit to me.

I think it would also be fine to move this `Headers` extension to a separate file as well if you are concerned about file length.